src/shared/interfaces/StateAbleInterface.tsx

```diff
@@ -0,0 +1,5 @@
+export interface StateAble<T> {
```

Comment:
Still I don't see good name :P

Reply:
FetchAble maybe? This interface will be used when doing fetch calls where we need loading, error and data in state.

Reply:
But in almost every place we need error and loading so I think we could create separate interface for isLoading and isError